[DCP][BC] Add the backward compatibility test

fegin opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #116248
* __->__ #116247
* #116246
* #116245

This PR adds a test to ensure all metadata is backward compatible with the older definination.

Differential Revision: [D52357733](https://our.internmc.facebook.com/intern/diff/D52357733/)

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @LucasLLC

fegin committed:
[DCP][BC] Add the backward compatibility test

This PR adds a test to ensure all metadata is backward compatible with the older definination.

Differential Revision: [D52357733](https://our.internmc.facebook.com/intern/diff/D52357733/)

[ghstack-poisoned]

fegin committed:
Update on "[DCP][BC] Add the backward compatibility test"


This PR adds a test to ensure all metadata is backward compatible with the older definination.

Differential Revision: [D52357733](https://our.internmc.facebook.com/intern/diff/D52357733/)

[ghstack-poisoned]

wz337 reviewed (approved):
Thanks for adding this! LGTM

fegin committed:
Update on "[DCP][BC] Add the backward compatibility test"


This PR adds a test to ensure all metadata is backward compatible with the older definination.

Differential Revision: [D52357733](https://our.internmc.facebook.com/intern/diff/D52357733/)

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 LucasLLC

[ghstack-poisoned]

fegin committed:
Update on "[DCP][BC] Add the backward compatibility test"


This PR adds a test to ensure all metadata is backward compatible with the older definination.

Differential Revision: [D52357733](https://our.internmc.facebook.com/intern/diff/D52357733/)

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 LucasLLC

[ghstack-poisoned]

fegin committed:
Update on "[DCP][BC] Add the backward compatibility test"


This PR adds a test to ensure all metadata is backward compatible with the older definination.

Differential Revision: [D52357733](https://our.internmc.facebook.com/intern/diff/D52357733/)

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 LucasLLC

[ghstack-poisoned]
